gemcalibration/src/common/Calibration.cc

```diff
@@ -0,0 +1,398 @@
+        if (it.first.find("armDac")!=std::string::npos || it.first.find("trimValues")!=std::string::npos) continue;
+        (calibConfigMap->find(it.first))->second =  std::stoi(it.second.c_str());
+        
+    }
+}
+
+void gem::calib::Calibration::fillCalibTypeConfigMap(calType_t cal, std::map<std::string, xdata::String>* calibTypeConfigMap) {
+    
+    for ( auto it: m_scanParams.find(cal)->second ) {
+        if (it.first.find("armDac")!=std::string::npos || it.first.find("trimValues")!=std::string::npos) {
+         calibTypeConfigMap->find(it.first)->second = it.second;
+        }
+    }
+}
+
+void gem::calib::Calibration::printCalibConfigMap(  std::map<std::string, xdata::Integer>* calibConfigMap ) {
+     std::map<std::string,xdata::Integer>::iterator iter;
+     CMSGEMOS_DEBUG("Calibration::printCalibConfigMap  Printing CalibConfigMap");
+     for ( iter = calibConfigMap->begin(); iter != calibConfigMap->end() ;++iter ) {
+         CMSGEMOS_DEBUG(" iter->first "<<iter->first << " second " << iter->second.toString());
+    }
+}
+       
+ void gem::calib::Calibration::fillBagFromConfigMap( std::unordered_map<std::string, xdata::Serializable*>* bag, std::map<std::string, xdata::Integer>* calibConfigMap, std::map<std::string, xdata::String>* calibTypeConfigMap) {
+    
+    std::map<std::string,xdata::Integer>::iterator iter;
+    for ( iter = calibConfigMap->begin(); iter != calibConfigMap->end() ;++iter ) {
+        bag->insert(std::make_pair(iter->first,    &(iter->second)));
+    }
+    std::map<std::string,xdata::String>::iterator iterStr;
+    for ( iterStr = calibTypeConfigMap->begin(); iterStr != calibTypeConfigMap->end() ;++iterStr ) {
+        bag->insert(std::make_pair(iterStr->first,    &(iterStr->second)));  
+    }
+}
+
+void gem::calib::Calibration::initializeAndFillDacScanConfigMap(std::map<std::string, dacFeatureBag>* dacScanConfigMap) {
+    for (auto it :  m_dacScanTypeParams ) {  
+            
+        
+        if (dacScanConfigMap->find(it.second.label)==dacScanConfigMap->end()) {
+            dacFeatureBag tmp;
+            tmp.label= it.second.label;
+            tmp.min=it.second.min;
+            tmp.max=it.second.max;
+            tmp.scan=it.second.scan;
+            dacScanConfigMap->insert( std::pair <std::string,dacFeatureBag> (it.second.label, tmp ) );
+            
+        }else{
+            dacScanConfigMap->find(it.second.label)->second.min = it.second.min;
+            dacScanConfigMap->find(it.second.label)->second.max = it.second.max;
+            dacScanConfigMap->find(it.second.label)->second.scan = it.second.scan;
+        }           
+        
+    }
+}
+
+void gem::calib::Calibration::fillDacScanBagFromConfigMap( std::unordered_map<std::string, xdata::Serializable*>* bag, std::map<std::string, dacFeatureBag>* dacScanConfigMap ) {
+    
+    std::map<std::string, dacFeatureBag>::iterator iter;
+    for ( iter = dacScanConfigMap->begin(); iter != dacScanConfigMap->end() ;++iter ) {
```

Comment:
range-based loops should be used throughout the code. Now all our code can benefit of relatively modern c++ instead of elderly c++0x :) 
Moreover, at l272 of the same file range-based `for` loop is used. We need to be at least consistent.